### PR TITLE
MB-13773 Pre-work to consolidate some definitions before re-using them for the actual ticket work

### PR DESCRIPTION
**Pull request by felipe-lee**
## Summary

This PR isn't the work to fulfill the ticket AC. It's more pre-work to re-use definitions across the internal and ghc APIs which will then be re-used for the new definitions I'll be adding to meet the AC.

I moved the Upload, Document, and Weight Ticket payload definitions to be shared between the internal and ghc apis. Had to rename `UploadPayload` to `Upload` and `DocumentPayload` to `Document` because [our definitions shouldn't end in `Payload`](https://github.com/transcom/mymove/pull/9323#discussion_r998430059).

The weight ticket definition was already being re-used in practice, but lived in the internal.yaml so I moved it to a shared file.

One thing to note is that there are two shared Upload definitions, one is used by internal and ghc while the other is used by the prime and support apis. I took a brief look at aligning them all, but the prime and support api definition is more loose (fewer required fields) and the corresponding BE code is also more forgiving about data missing, so I didn't want to rework all of that in this PR (nor do I know if it needs to be that loose, which would take some investigation). The stricter definition will use the name `Upload` (internal & ghc) while the more loose one will be called `UploadWithOmissions` (prime & support) since it allows for fields to be omitted (like the URL). 

The internal and ghc APIs already used definitions that were nearly identical for Upload and Document, the only differences being that one use camel case while the other used snake case, and one had an additional `checksum` field being returned as part of the upload payload, but nothing used that field. I aligned them both on using camel case on the Upload definition, but left the snake case on the Document payload since the field that is snake cased is the `service_member_id`, which has widespread use and would take more time to fix. As for the `checksum` field, I just got rid of it at the API level since it doesn't matter at that level. It still exists in the rest of the BE code though.

Note that Design has been tagged because stories got updated, but there are no visual changes because the changes were only to variable names.

### Local Setup

<details>
<summary>💻</summary>

#### Terminal 1

Start the Storybook locally.

```sh
make storybook
```

#### Terminal 2

Run FE tests.

```sh
make client_test
```

#### Terminal 3

Run BE tests.

```sh
make server_test
```

</details>

### Additional steps

1. Once storybook is up, you should be able to navigate to stories that have document upload bits like the `UploadsTable` component.
    1. Alternatively, you can use the [storybook built by CI](https://output.circle-artifacts.com/output/job/6b4894b0-fd0e-4af3-bd47-d24b82af4f73/artifacts/0/storybook/index.html).
1. There shouldn't be any visible differences, they were only variable name changes.

### CODE REVIEW

**Review by rileyorr — APPROVED** [review]
Thank you for tagging me about the stories being updated! I took a look at the stories in Storybook and reviewed the Happo diffs.

**Review by ronaktruss — APPROVED** [review]
LGTM! Thanks for cleaning up some of propTypes 🥳. Left some comments that are probably nice to haves but these changes are on existing code that is being relocated and not explicitly something that you have changed.

File: swagger-def/definitions/WeightTicket.yaml [hunk 1]
```diff
@@ -0,0 +1,96 @@
+description: Vehicle and optional trailer information and weight documents used to move this PPM shipment.
+type: object
+properties:
```
[the remark this answers is not in the document]
Comment (felipe-lee) [reply swagger-def/definitions/WeightTicket.yaml:3]: I copied over the DB comments, with some tweaks (like removing "stores x" type language).

File: swagger-def/definitions/WeightTicket.yaml [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    type: string
+    format: date-time
+    readOnly: true
+  vehicleDescription:
+    type: string
+    title: Vehicle description (ex. 'SUV')
+    x-nullable: true
+  emptyWeight:
+    type: integer
+    title: Empty Recorded Weight
```
[the remark this answers is not in the document]
Comment (felipe-lee) [reply swagger-def/definitions/WeightTicket.yaml]: Done!

File: swagger-def/definitions/WeightTicket.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (felipe-lee) [reply swagger-def/definitions/WeightTicket.yaml:3]: I also fixed a few things that weren't working how we wanted them to (e.g. the document definitions that weren't using `allOf`).

File: swagger-def/definitions/WeightTicket.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (duncan-truss) [reply swagger-def/definitions/WeightTicket.yaml]: Yea titles don't display that prominently in the Redoc output like descriptions do and sometimes add confusion.

File: swagger-def/definitions/MovingExpense.yaml [hunk 3]
```diff
@@ -20,12 +20,11 @@ properties:
     example: c56a4180-65aa-42ec-a945-5fd21dec0538
     readOnly: true
   document:
-    description: The Document object that contains all file uploads for this expense
-    type: object
+    allOf:
+      - description: The Document object that contains all file uploads for this expense
+      - $ref: 'DocumentPayload.yaml'
```
Comment (duncan-truss) [inline swagger-def/definitions/MovingExpense.yaml:25]: Naming something Payload will get it flagged by the Spectral linter https://github.com/transcom/mymove/blob/master/swagger-def/.spectral.yml#L141

File: swagger-def/definitions/OmittableMovingExpenseType.yaml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+enum:
+  - CONTRACTED_EXPENSE
+  - GAS
+  - OIL
+  - OTHER
+  - PACKING_MATERIALS
+  - RENTAL_EQUIPMENT
+  - STORAGE
+  - TOLLS
+  - WEIGHING_FEES
```
Comment (duncan-truss) [inline swagger-def/definitions/OmittableMovingExpenseType.yaml:12]: Heads up Scott's PR changes this value so there will likely be conflicts for one of you.

File: swagger-def/definitions/MovingExpense.yaml (hunk 3 above)
Comment (felipe-lee) [reply swagger-def/definitions/MovingExpense.yaml:25]: Thanks for calling this out. I'll run the linter and make fixes

Comment (ronaktruss) [reply swagger-def/definitions/MovingExpense.yaml]: Thanks for cleaning these up, they look good to me.

**Pull request merged**